clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -844,29 +821,18 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
                 throw new IllegalStateException("Unexpected error code " + error.code() + " while fetching data");
             }
         } finally {
-            completedFetch.metricAggregator.record(tp, bytes, recordsCount);
+            if (error != Errors.NONE) {
+                completedFetch.metricAggregator.record(tp, bytes, recordsCount);
```

Comment:
Could we just call `record(tp, 0, 0)` here and get rid of the `bytes` and `recordsCount` variables?